DNN Platform/DotNetNuke.Maintenance/Telerik/TelerikUtils.cs

```diff
@@ -1,128 +1,136 @@
+        };
+
+        private static readonly string[] WellKnownAssemblies = new[]
+        {
+            "Telerik.Web.UI.Skins.dll",
+            "DotNetNuke.Website.Deprecated.dll",
+            "DotNetNuke.Web.Deprecated.dll",
+            "DotNetNuke.Modules.DigitalAssets.dll",
+        };
+
+        private readonly IApplicationStatusInfo applicationStatusInfo;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TelerikUtils"/> class.
+        /// </summary>
+        /// <param name="applicationStatusInfo">
+        /// An instance of <see cref="IApplicationStatusInfo"/>.
+        /// </param>
+        public TelerikUtils(IApplicationStatusInfo applicationStatusInfo)
+        {
+            this.applicationStatusInfo = applicationStatusInfo
+                ?? throw new ArgumentNullException(nameof(applicationStatusInfo));
+        }
+
+        /// <inheritdoc />
+        public string BinPath => Path.Combine(this.applicationStatusInfo.ApplicationMapPath, "bin");
+
+        /// <inheritdoc/>
+        public IEnumerable<string> GetAssembliesThatDependOnTelerik()
+        {
+            // use a temp AppDomain to avoid locking files in the bin folder
+            var domain = AppDomain.CreateDomain(nameof(TelerikUtils));
+            try
+            {
+                return this.DirectoryGetFiles(this.BinPath, "*.dll", SearchOption.AllDirectories)
+                    .Where(this.IsNotWellKnownAssembly)
+                    .Where(path => this.AssemblyDependsOnTelerik(path, domain))
+                    .ToList();
+            }
+            finally
+            {
+                AppDomain.Unload(domain);
+            }
+        }
+
+        /// <inheritdoc />
+        public bool TelerikIsInstalled()
+        {
+            return this.FileExists(Path.Combine(this.BinPath, TelerikWebUIFileName));
+        }
+
+        /// <inheritdoc />
+        public Version GetTelerikVersion()
+        {
+            var domain = AppDomain.CreateDomain(nameof(TelerikUtils));
+            try
+            {
+                var path = Path.Combine(this.BinPath, TelerikWebUIFileName);
+                return domain.Load(File.ReadAllBytes(path)).GetName().Version;
+            }
+            catch (Exception ex)
+            {
+                throw new IOException("Could not get Telerik version number.", ex);
+            }
+            finally
+            {
+                AppDomain.Unload(domain);
+            }
+        }
+
+        private bool AssemblyDependsOnTelerik(string path, AppDomain domain)
+        {
+            try
+            {
+                return this.GetReferencedAssemblyNames(path, domain)
+                    .Any(assemblyName => assemblyName.StartsWith("Telerik", StringComparison.OrdinalIgnoreCase));
+            }
+            catch (Exception)
+            {
+                // If we can't get the referenced assemblies, then it can't depend on Telerik.
+                return false;

```

Comment:
Ideally I would love to see this logged, so they can at least know. As it is not ALWAYS that it cannot be loaded. It could be file in use. Etc.

Reply:
Does https://github.com/dnnsoftware/Dnn.Platform/pull/5269/commits/0e70665f658cfa6ef3e62ea55b10b5bdebe8e509 work ? If we want to surface this to the UI, we may have to reachitect that...

Reply:
That would be good enough for me!

Reply:
My goal isn't to make it crazy, but for when things go sideways, we can direct them to the /portals/_default/logs folder